### DISCUSSION TITLE
Is machine a library or a module?

### DISCUSSION THREAD

**Discussion by retrobrian** [micropython/micropython#16062] [Libraries & Drivers] [answered]
Hello, 
When I read the micropython.org documentation, "machine" is listed as a MicroPython specific **library** on this page: https://docs.micropython.org/en/latest/library/index.html

But, when I view the page for "machine", here: https://docs.micropython.org/en/latest/library/machine.html,

I find this statement, "The **machine** module contains specific functions..."

Some I'm wondering, is machine considered a library or a module?

Thank you!

-Brian

**Comment by mattytrentini** [comment, marked as answer]
In Python, the terms _module_, _package_ and _library_ are often used somewhat interchangeably, though they do have technical differences as described here: [Python Modules, Packages, Libraries, and Frameworks](https://learnpython.com/blog/python-modules-packages-libraries-frameworks/). People will also argue about the definitions.

`machine` is probably most appropriately called a _library_ but I would also use the term _module_. I wouldn't sweat it too much. 😜

**Reply by retrobrian** [reply]
Thank you, that is exactly what I needed to know. I'm teaching MicroPython to some kids and I don't want to confuse them with minutiae if I don't have to!
